experimental/sql/dialects/ibis_dialect.py

```diff
@@ -0,0 +1,247 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from typing import Any
+from xdsl.ir import Block, Region, Operation, SSAValue, ParametrizedAttribute, Data, MLContext, Attribute
+from xdsl.dialects.builtin import StringAttr, ArrayAttr, ArrayOfConstraint, IntegerAttr
+from xdsl.irdl import AttributeDef, OperandDef, ResultDef, RegionDef, SingleBlockRegionDef, irdl_attr_definition, irdl_op_definition, ParameterDef, AnyAttr, VarOperandDef, builder
+
+# This file defines a dialect that mirrors the internal data
+# structures of the ibis framework (https://ibis-project.org/docs/dev/).
+# DataTypes are defined as Attributes, other important nodes are
+# operations. Everything that is an `expresssion` in ibis is not
+# modelled.
+
+
+@irdl_attr_definition
+class DataType(ParametrizedAttribute):
+  """
+  Models an ibis datatype. 
```

Comment:
There is a trailing white space in this and several other lines. The simplest is maybe to set `files.trimTrailingWhitespace": true` in `settings.json`.